docs/core/diagnostics/logging-tracing.md

```diff
@@ -0,0 +1,94 @@
+---
+title: .NET Core Logging and Tracing
+description: An introduction to .NET Core Logging and Tracing.
+author: sdmaclea
+ms.author: stmaclea
+ms.date: 08/05/2019
+---
+# .NET Core Logging and Tracing
+
+Logging and tracing is a primitive technique used since the early days of computers. It simply involves instrumenting an application to produce output to be consumed later.
+
+Logging and tracing are similar. As used here we'll distinguish:
+- Logging outputs text as human readable content.
+- Tracing outputs binary content.  It can use tokenizing, and filtering to reduce overhead in the traced application.
+
+## Why Logging and Tracing?
+
+These simple techniques are surprisingly powerful. They can be used in situations where a fancy debugger fails:
+
+- Issues occurring over long periods of time, can be difficult to debug with a traditional debugger.
+- Multi-threaded applications and distributed applications are often difficult to debug.  Attaching a debugger tends to modify behaviors. Detailed logs can be analyzed as needed to understand complex systems.
+- Many services shouldn't be stalled. Attaching a debugger often causes timeout failures.
+
+## Logging APIs
+
+<xref:System.Console?displayProperty=nameWithType>, <xref:System.Diagnostics.Trace?displayProperty=nameWithType>, and <xref:System.Diagnostics.Debug?displayProperty=nameWithType> each provide similar APIs convenient for logging.
+
+The choice of which API to use is up to you. The key differences:
+- <xref:System.Console?displayProperty=nameWithType>
+  - Always compiled in and always writes to the console.
+  - Useful for information that your customer may need to see in release.
+  - Because it's the simplest approach, it's often used for ad-hoc temporary debugging. This debug code is often never checked in to source control.
+- <xref:System.Diagnostics.Trace?displayProperty=nameWithType>
+  - Only enabled when `TRACE` is defined.
+  - Writes to attached <xref:System.Diagnostics.TraceListener>s. Usually a debugger.
```

Comment:
I'm not sure what the "Usually a debugger" means?

Reply:
Don't make the plural by attaching the s because it's bad for loc. Define what the TraceListener is. Object?

```suggestion
  - Writes to attached <xref:System.Diagnostics.TraceListener> objects. Usually a debugger.
```